core/variant/variant.cpp

```diff
@@ -1291,7 +1291,13 @@ void Variant::zero() {
 			break;
 
 		default:
+			Type prev_type = type;
 			this->clear();
+			if (type != prev_type) {
+				// clear() changes type to NIL, so it needs to be restored.
```

Comment:
Might be slighter better to add a `bool reset_type = true` to `clear()` instead of doing it like this?

Reply:
`_clear_internal()` called by `clear()` will free the internal data, so it needs to be re-created.
The best solution would be probably reworking `zero()` to not use `clear()`.